Model/Api.php

```diff
@@ -383,12 +383,13 @@ public function getCustomerInfo()
      * @param bool $test
      * @param null $serviceId
      * @param null $apiKey
+     * @param bool $isInitialCheck - flag for testing credentials before Service ID and token are saved to config
      * @return bool|mixed
      * @throws LocalizedException
      */
-    public function checkServiceDetails($test = false, $serviceId = null, $apiKey = null)
+    public function checkServiceDetails($test = false, $serviceId = null, $apiKey = null, $isInitialCheck = false)
     {
-        if (!$this->config->isServiceConfigured()) {
+        if (!$this->config->isServiceConfigured() && !$isInitialCheck) {
```

Comment:
Testing a (boolean) variable for truthiness is cheaper than calling a method on an object. It would be more efficient to put the less expensive check first.
```suggestion
        if (!$isInitialCheck && !$this->config->isServiceConfigured()) {
```